app/controllers/application_controller.rb

```diff
@@ -124,4 +125,23 @@ def pull_categories
       Category.all.order(sequence: :asc, id: :asc)
     end
   end
+
+  def check_if_warning_or_suspension_pending
+    return if current_user.nil?
+
+    warning = ModWarning.where(community_user: current_user.community_user, active: true).any?
+    return unless warning
+
+    path = Rails.application.routes.recognize_path(request.path, method: request.env['REQUEST_METHOD'])
+
+    # Ignore devise and warning routes
+    return if path[:controller] == 'custom_sessions'
+    return if path[:controller] == 'users' && path[:action] == 'show' && path[:id] == 'sign_out'
+    return if path[:controller] == 'mod_warning'
+    return if path[:controller] == 'errors'
```

Comment:
Rails has a magic `controller_name` variable that you can use here:

```suggestion
    return if devise_controller? || ['custom_sessions', 'mod_warning', 'errors'].include?(controller_name)
```